ObjectCreation/boxcreator/boxcreator.py

```diff
@@ -123,11 +124,14 @@ def draw_bottom(partname, materialWidth, boxWidth, boxLength, notchWidth, drawSi
     Draft.move(lines4, FreeCAD.Vector(0.0, boxLength, 0.0), copy=False)
     lines.extend(lines4)
 
+    doc = FreeCAD.activeDocument()
```

Comment:
Please add a check that a document is open (i.e. FreeCAD.activeDocument() is not None).

According to me, the best would be to add a `doc` argument to `create_box` and manage getting the current document (or possibly create a new one) in `BoxCreator.FCMacro`. This would allow someone to use the `create_box` function in scripts with multiple documents without having to take care about the fact whether the document is active or not.